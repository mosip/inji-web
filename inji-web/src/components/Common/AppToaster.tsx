import React from "react";
import {ToastContainer} from "react-toastify";

export const AppToaster: React.FC = () => {
    return <ToastContainer
        position="top-right"
        autoClose={1000}
        hideProgressBar
        newestOnTop
        closeOnClick
        rtl={false}
        icon={<React.Fragment/>}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        style={{width: '400px'}}
        theme="colored"
    />
}
