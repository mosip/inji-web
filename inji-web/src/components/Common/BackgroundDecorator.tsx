import React from 'react';

interface BackgroundDecoratorProps {
    circleCount?: number;
    topOffset?: number;
    baseRadius?: number;
    logoSrc?: string;
    logoAlt?: string;
    testId?: string;
}

export const BackgroundDecorator: React.FC<BackgroundDecoratorProps> = ({
    circleCount = 6,
    topOffset = 155,
    baseRadius = 96,
    logoSrc,
    logoAlt = "Logo",
    testId = "background-decorator"
}) => {
    return (
        <div 
            className="overflow-hidden absolute inset-0 z-0 flex items-center justify-center pointer-events-none"
            data-testid={testId}
        >
            <div className={`absolute top-[${topOffset}px]`} style={{ top: `${topOffset}px` }}>
                {[...Array(circleCount)].map((_, index) => {
                    const radius = baseRadius + index * baseRadius;
                    const opacity = 0.8 - index * 0.1;
                    return (
                        <div
                            key={index}
                            className="absolute rounded-full border overflow-hidden"
                            style={{
                                width: `${radius}px`,
                                height: `${radius}px`,
                                borderWidth: '1px',
                                borderColor: `rgba(228, 231, 236, ${opacity})`,
                                top: `calc(50% - ${radius / 2}px)`,
                                left: `calc(50% - ${radius / 2}px)`
                            }}
                            data-testid={`${testId}-circle-${index}`}
                        />
                    );
                })}
                
                {logoSrc && (
                    <div
                        className="flex items-center justify-center"
                        data-testid={`logo-${testId}-container`}
                    >
                        <img
                            src={logoSrc}
                            alt={logoAlt}
                            data-testid={`logo-${testId}`}
                        />
                    </div>
                )}
            </div>
        </div>
    );
};