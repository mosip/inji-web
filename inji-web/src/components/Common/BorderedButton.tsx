import React from "react";

export const BorderedButton:React.FC<BorderedButtonProps> = (props) => {
  return <div className={"bg-gradient-to-r rounded-lg from-iw-primary to-iw-secondary p-0.5"}>
            <div className="py-2 px-4 rounded-lg bg-white">
                <button
                    data-testid={props.testId}
                    onClick={props.onClick}
                    className="text-iw-tertiary font-bold rounded-lg">
                    {props.title}
                </button>
            </div>
    </div>
}

export type BorderedButtonProps = {
    testId: string;
    onClick: ()=> void;
    title: string;
}
