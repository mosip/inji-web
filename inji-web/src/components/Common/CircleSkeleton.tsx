type CircleSkeletonProps = {
    size?: string;
    className?: string;
};

export const CircleSkeleton = ({
                                   size = 'w-20 h-20',
                                   className = '',
                               }: CircleSkeletonProps) => {
    return (
        <div
            className={`rounded-full bg-gray-300 animate-pulse ${size} ${className}`}
        />
    );
};
  