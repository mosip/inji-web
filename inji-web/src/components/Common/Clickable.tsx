import React from 'react';

export interface ClickableProps {
    onClick: () => void;
    children: React.ReactNode;
    testId: string;
    className: string;
}

export const Clickable: React.FC<ClickableProps> = ({
                                                        onClick,
                                                        children,
                                                        testId,
                                                        className = '',
                                                    }) => {
    const handleKeyPress = (event: React.KeyboardEvent<HTMLDivElement>) => {
        if (event.key === 'Enter' || event.key === ' ') {
            event.preventDefault();
            onClick();
        }
    };

    return (
        <div
            role="menuitem"
            tabIndex={0}
            className={className}
            onClick={onClick}
            onKeyDown={handleKeyPress}
            data-testid={testId}
        >
            {children}
        </div>
    );
};
