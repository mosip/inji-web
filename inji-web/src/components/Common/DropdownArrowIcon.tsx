import React from 'react';
import {RiArrowUpSFill, RiArrowDownSFill} from 'react-icons/ri';
import { DropdownArrowIconProps } from '../../types/data';
import { GradientWrapper } from './GradientWrapper';

const DropdownArrowIcon: React.FC<DropdownArrowIconProps> = ({isOpen, size = 20, color = 'var(--iw-color-languageArrowIcon)'}) => {
    const ArrowIcon = isOpen ? RiArrowUpSFill : RiArrowDownSFill;

    return (
        <GradientWrapper>
            <ArrowIcon size={size} color={color} />
        </GradientWrapper>
    );
};

export default DropdownArrowIcon;
