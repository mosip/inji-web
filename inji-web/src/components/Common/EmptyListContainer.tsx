import React from "react";
import {EmptyListContainerProps} from "../../types/components";

export const EmptyListContainer: React.FC<EmptyListContainerProps> = ({content}) => {
    return <React.Fragment>
        <div data-testid="EmptyList-Outer-Container"
             className="flex justify-center items-center w-full mx-auto flex-col">
            <div data-testid="EmptyList-Inner-Container" className="container mx-auto mt-8 px-4 flex-1 flex flex-col">
                <p data-testid="EmptyList-Text" className="text-center">{content}</p>
            </div>
        </div>
    </React.Fragment>
}
