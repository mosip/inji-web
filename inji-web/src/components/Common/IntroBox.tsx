import React from "react";
import {useTranslation} from "react-i18next";

export const IntroBox: React.FC = () => {
    const {t} = useTranslation("HomePage");
    return <React.Fragment>
        <div data-testid="IntroBox-Container" className="text-center pt-20 pb-20">
            <h2 data-testid="IntroBox-Text"
                className="text-3xl text-iw-title ">{t("Intro.title")}</h2>
            <p data-testid="IntroBox-SubText"
               className="mt-2 text-lg text-iw-subTitle">{t("Intro.subTitle")}</p>
        </div>
    </React.Fragment>
}
