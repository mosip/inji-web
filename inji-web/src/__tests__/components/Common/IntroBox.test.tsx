import React from 'react';
import { render, screen } from '@testing-library/react';
import {IntroBox} from "../../../components/Common/IntroBox";



describe("Test Intro Box Layout",() => {
    test('check the presence of the container', () => {
        render(<IntroBox />);
        const introBoxElement = screen.getByTestId("IntroBox-Container");
        expect(introBoxElement).toBeInTheDocument();
    });
    test('check the presence of the title', () => {
        render(<IntroBox />);
        const introBoxElement = screen.getByTestId("IntroBox-Text");
        expect(introBoxElement).toBeInTheDocument();
    });
    test('check the presence of the subTitle', () => {
        render(<IntroBox />);
        const introBoxElement = screen.getByTestId("IntroBox-SubText");
        expect(introBoxElement).toBeInTheDocument();
    });
})


describe("Test Intro Box Content",() => {
    test('check if content is rendered properly', () => {
        render(<IntroBox />);
        const headerElement = screen.getByTestId("IntroBox-Text");
        expect(headerElement).toHaveTextContent("Intro.title")
    });
    test('check if content is rendered properly', () => {
        render(<IntroBox />);
        const headerElement = screen.getByTestId("IntroBox-SubText");
        expect(headerElement).toHaveTextContent("Intro.subTitle")
    });
})
