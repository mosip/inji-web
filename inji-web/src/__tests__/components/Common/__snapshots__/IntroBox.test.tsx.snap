// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of IntroBox Check if the layout is matching with the snapshots 1`] = `
<DocumentFragment>
  <div
    class="text-center pb-10"
    data-testid="IntroBox-Container"
  >
    <h2
      class="text-lg sm:text-xl lg:text-2xl font-medium text-iw-title "
      data-testid="IntroBox-Text"
    >
      Intro.title 
      <span
        class="bg-gradient-to-r from-iw-primary to-iw-secondary bg-clip-text text-md font-bold text-transparent"
      >
        Intro.title2
      </span>
    </h2>
    <p
      class="mt-2 sm:text-md lg:text-lg font-small text-iw-subTitle"
      data-testid="IntroBox-SubText"
    >
      Intro.subTitle
    </p>
  </div>
</DocumentFragment>
`;
